repair/row_level.cc

```diff
@@ -1866,14 +1909,22 @@ class repair_meta {
     }
 };
 
-void flush_rows(schema_ptr s, std::list<repair_row>& rows, lw_shared_ptr<repair_writer>& writer, locator::effective_replication_map_ptr erm, bool small_table_optimization) {
+void flush_rows(schema_ptr s, std::list<repair_row>& rows, lw_shared_ptr<repair_writer>& writer, locator::effective_replication_map_ptr erm, bool small_table_optimization, repair_meta* rm) {
     auto cmp = position_in_partition::tri_compare(*s);
     lw_shared_ptr<mutation_fragment> last_mf;
     lw_shared_ptr<const decorated_key_with_hash> last_dk;
     bool do_small_table_optimization = erm && small_table_optimization;
     auto* strat = do_small_table_optimization ? &erm->get_replication_strategy() : nullptr;
-    auto* tm = do_small_table_optimization ? &erm->get_token_metadata() : nullptr;
+    const auto* tm = do_small_table_optimization ? &erm->get_token_metadata() : nullptr;
     auto myip = do_small_table_optimization ? erm->get_topology().my_address() : gms::inet_address();
+
+    if (do_small_table_optimization && rm) {
+        const auto& tmptr = rm->get_tm_for_small_table_optimization_check(tm);
```

Comment:
Looks like we never get here if `!tm`, why is this case still handled?

Reply:
Yes, currently we always pass a valid tm to get_tm_for_small_table_optimization_check. Checking !tm in  get_tm_for_small_table_optimization_check is just for being defensive.